sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamTest.scala

```diff
@@ -429,7 +429,25 @@ trait StreamTest extends QueryTest with SharedSQLContext with TimeLimits with Be
     val defaultCheckpointLocation =
       Utils.createTempDir(namePrefix = "streaming.metadata").getCanonicalPath
     try {
-      startedTest.foreach { action =>
+      val actionIterator = startedTest.iterator.buffered
+      while (actionIterator.hasNext) {
+        // Synchronize sequential addDataMemory actions.
```

Comment:
// Synchronize --> // Collectively synchronize .... actions so that the data gets added together in a single batch.